### ISSUE TITLE
Formatter: mixed visibility constants in same segment

### ISSUE THREAD

**Issue by MakeNowJust** [crystal-lang/crystal#4113]
For example:

```crystal
private Foo = 1
A = 2
```

How is this file formatted?

I consider 2 behaviors:

1.  ```crystal
    private Foo = 1
            A   = 2
    ```

2. ```crystal
   private Foo = 1
   A           = 2
   ```

1. is pretty good, but this implementation will be heavy, I think. 2. is not the best, but we can implement it easily.

Which do you like?  Or do you have any idea?

**Comment by Bew78LesellB** [comment]
I would keep the first one, for me the 2 should not be in the same segment, and maybe the formatter should even separate them. 

The visibility is different, the formatting should clearly show that:
```crystal
private Foo = 1
protected Bar = 4
protected FooBar = 5
Fizz = 10
```
For me should be formatted as:
```crystal
private Foo = 1

protected Bar    = 4
protected FooBar = 5

Fizz = 10
```

**Comment by ysbaddaden** [comment]
I wish constants weren't aligned on their assignment. We'd have a nicer-to-read-patch-diff.
```diff
  Short = 1
- VeryLongConstant = 2
+ Renamed = 2
```

Instead of:
```diff
- Short            = 1
- VeryLongConstant = 2
+ Short   = 1
+ Renamed = 2
```

Then @bew's example would lead to no change by the formatter:
```crystal
private Foo = 1
protected Bar = 4
protected FooBar = 5
Fizz = 10
```

**Comment by bcardiff** [comment]
@ysbaddaden unless we switch to elastic-tabstops I think those extra diff lines are things we will need to ~deal with~ live with. And the editors haven't catch up yet to use elastic-tabstops, I don't see that changing any time soon.

**Comment by MakeNowJust** [comment]
I have more general question.

Should the following segment be separated?

```crystal
private property foo : String
private property bar : Int32
protected property fizz : String
protected property buzz : String
```

↓

```crystal
private property foo : String
private property bar : Int32

protected property fizz : String
protected property buzz : String
```

In other words, should we separate segments by visibility modifiers?

**Comment by bcardiff** [comment]
@MakeNowJust I don't think so.  I would aim for aligning constants only.

Otherwise, maybe:
1. assignments in properties for default values should be aligned also.
```
property foo = false
property longer_foo = false
```

2. segments with properties and plain ivars should be separated.
```
property foo
@bar = 42
property baz
```

currently we do nothing on those cases and I sometimes mix them (might not be the best thing to do, but is how my code sometimes evolve, putting related props/ivars together).

constants, since they are read only they don't "suffer" that use case.

**Comment by cluxter** [comment]
I have a strong preference for the first suggestion in the @MakeNowJust 's message over all the others for 2 reasons:
- we can easily read the types, the names and the values with no effort because not only you can read horizontally but also vertically. Each column has a meaning so you can easily review the consistency of the code.
- Since each column has a meaning, it's an improvement for security because you can compare lines more easily. The fact that you can read vertically makes debugging way faster for the eye. It almost appears more like an image for the brain than text that you must read line by line.

**Comment by MakeNowJust** [comment]
I agree with @cluxter's opinion. And, the formatter should not separate lines without user's intention in almost all case.

**Comment by RX14** [comment]
Personally I would like to see @bew's suggestion, judging by the reactions it has quite a bit of support too.

**Comment by sdogruyol** [comment]
What's the last stand on this?

**Comment by asterite** [comment]
My suggestion is that the compiler shouldn't align things, ever. It was a mistake.

**Comment by RX14** [comment]
Are diffs really an issue? On github at least we have `?w=1`.
